Helper/Data.php

```diff
@@ -210,4 +487,29 @@ public function isCustomerGroupAllowed()
         $allowedCustomerGroups = $this->balancepayConfig->getAllowedCustomerGroups();
         return in_array($currentCustomerGroup, $allowedCustomerGroups);
     }
+
+    /**
+     * @param $params
+     * @param $webhook
+     * @throws LocalizedException
+     * @throws \Magento\Framework\Exception\NoSuchEntityException
+     */
+    public function processWebhookCron($params, $webhook)
+    {
+        $isTransactionSuccess = false;
+        $order = $this->orderFactory->create()->loadByIncrementId((string)$params['externalReferenceId']);
+        if (!$order || !$order->getId()) {
+            $attempts = $webhook->getAttempts() + 1;
+            $this->updateAttempts($webhook->getEntityId(), 'attempts', $attempts);
+            throw new LocalizedException(new Phrase("No matching order!"));
+        }
+        if ($webhook->getName() == Confirmed::WEBHOOK_CONFIRMED_NAME) {
```

Comment:
Where is this enum defined? I think there should be one enum `BalanceWebhook`, that should include `TransactionConfirmed` and `TransactionCharged` (and any other webhook in the future

Reply:
We have defined a name of the webhook in the webhook_cron table.